test/test_dataloader.py

```diff
@@ -2033,6 +2034,24 @@ def clean_me(process, req_queue, res_queue):
 
         self.assertEqual(list(range(100)), actual)
 
+    @skipIfNoDill
+    def test_basic_mapdatapipe_threading(self):
+        def clean_me(process, req_queue, res_queue):
+            req_queue.put(communication.messages.TerminateRequest())
+            _ = res_queue.get()
+            process.join()
+        it = list(range(100))
+        numbers_dp = SequenceWrapper(it)
+        (process, req_queue, res_queue, _thread_local_datapipe) = communication.eventloop.SpawnThreadForDataPipeline(numbers_dp)
+
+        process.start()
+        local_datapipe = communication.map.QueueWrapper(
+            communication.protocol.MapDataPipeQueueProtocolClient(req_queue, res_queue))
+        actual = list(local_datapipe)
+        clean_me(process, req_queue, res_queue)
+
+        self.assertEqual([(x, x) for x in range(100)], actual)
```

Comment:
Worth noting that currently `getitem` returns both `(key, value)`. Should we just return `value`?